src/main/scala/com/advancedtelematic/treehub/http/TreeHubRoutes.scala

```diff
@@ -33,18 +38,25 @@ class TreeHubRoutes(tokenValidator: Directive0,
     new DeltaResource(nsExtract, deltaStorage, usageHandler).route
   }
 
+  val treehubExceptionHandler = handleExceptions(ExceptionHandler.apply {
+    case ex: SdkClientException if ex.getMessage.contains("Timeout on waiting") =>
```

Comment:
Don't know enough scala to know what's going on here, but seems odd you have to match the error by string (if that is what's happening here)

Reply:
we are matching on the type `SdkClientException` **and** msg, because `SdkClientException` could be other errors for which we should throw a 5xx unfortunately the s3 sdk doesn't distinguish

Reply:
Ah right! Gotcha 👍